baseapp/abci.go

```diff
@@ -68,7 +68,14 @@ func (app *BaseApp) InitChain(req abci.RequestInitChain) (res abci.ResponseInitC
 
 	// NOTE: We don't commit, but BeginBlock for block 1 starts from this
 	// deliverState.
-	return res
+	return abci.ResponseInitChain{
+		ConsensusParams: res.ConsensusParams,
+		Validators:      res.Validators,
+		// In the case of a new chain, AppHash will be empty. During an
+		// upgrade, it'll be the hash of the last committed block.
+		AppHash: app.LastCommitID().Hash,
```

Comment:
btw, we return an empty hash, of the hash of "" on a new chain?

Reply:
should be the hash of an empty string. correct me if im wrong @erikgrinaker, but I believe IAVL already does this.